### ISSUE TITLE
Wrong nodeselector label to build app

### ISSUE THREAD

**Issue by fquinino** [Shuffle/Shuffle#1338]
I'm creating a new workflow, and I have to active the app, which is OpenSearch in this scenario. As a result, a job is initiated on my Kubernetes cluster to build this app and push it to the repository. However, the issue arises with the nodeSelector in the YAML, as it is using an incorrect label to name the node.

![image](https://github.com/Shuffle/Shuffle/assets/72146942/48c0e48d-4c62-4dd6-be12-245dec61b5f4)

The default Kubernetes label for hostname is kubernetes.io/hostname.

I'm using the latest version of Shuffle.

How can I modify the job puller behavior to use a different label for nodeSelector for the app job builder?

**Comment by frikky** [comment]
Local K8s man to the rescue: @dhaval055 please help our friend here :)

**Comment by fquinino** [comment]
Hello @frikky,

I'm using Rancher RKE on VMware. The issue I'm encountering is not related with local K8S only, it's related with the label you're using to assign the POD to host. According to the official Kubernetes documentation, the standard label for nodes is `kubernetes.io/hostname`.

![image](https://github.com/Shuffle/Shuffle/assets/72146942/4b883f3d-72d6-4767-a48d-4d4010c34ad6)

In your automation, you're using a label called `node` for the selector, which is not a standard Kubernetes label. You can refer to the official Kubernetes documentation for labels, annotations, and taints here: [Kubernetes Documentation](https://kubernetes.io/docs/reference/labels-annotations-taints/)

For testing purposes, I've temporarily solved the issue by tagging my node with the label `node` and setting its value to the be the hostname of my host. However, this is not good solution, right?. I suggest reviewing your application's design patterns to align with well-known Kubernetes labels. Otherwise, users will need to apply workarounds in their clusters.

Additionally, if you look at AKS or EKS, or the default labels in any Kubernetes node launched in AWS or Azure, you'll see that the `node` doesn't came as well.

My suggestion is Instead of using:

```yaml
nodeSelector:
  node: <name>
```

You should consider using:
```yaml
nodeSelector:
  kubernetes.io/hostname: <name>
```

Hope that clarify.

**Comment by dhaval055** [comment]
> Hello @frikky,
> 
> I'm using Rancher RKE on VMware. The issue I'm encountering is not related with local K8S only, it's related with the label you're using to assign the POD to host. According to the official Kubernetes documentation, the standard label for nodes is `kubernetes.io/hostname`.
> 
> ![Screenshot 2024-02-25 at 02 13 53](https://private-user-images.githubusercontent.com/72146942/307568584-439bf984-17d2-4e44-bf2e-8f83410e4a88.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg5NDA3MDYsIm5iZiI6MTcwODk0MDQwNiwicGF0aCI6Ii83MjE0Njk0Mi8zMDc1Njg1ODQtNDM5YmY5ODQtMTdkMi00ZTQ0LWJmMmUtOGY4MzQxMGU0YTg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI2VDA5NDAwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4ZWNlMWZmNDQ2MjkyMGM2Nzc0Yzc2YTRhMTU0ZWQ0NjllYzczNzBiNzcwOTRiYzE4ZGM1ZjFkZmI5MmJlY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bO7XAUceGWBGNbnfw-5r3ubj_EjdYN3fAK_n1G1DZos)
> 
> In your automation, you're using a label called `node` for the selector, which is not a standard Kubernetes label. You can refer to the official Kubernetes documentation for labels, annotations, and taints here: [Kubernetes Documentation](https://kubernetes.io/docs/reference/labels-annotations-taints/)
> 
> For testing purposes, I've temporarily solved the issue by tagging my node with the label `node` and setting its value to the be the hostname of my host. However, this is not good solution, right?. I suggest reviewing your application's design patterns to align with well-known Kubernetes labels. Otherwise, users will need to apply workarounds in their clusters.
> 
> Additionally, if you look at AKS or EKS, or the default labels in any Kubernetes node launched in AWS or Azure, you'll see that the `node` doesn't came as well.
> 
> My suggestion is Instead of using:
> 
> ```yaml
> nodeSelector:
>   node: <name>
> ```
> 
> You should consider using:
> 
> ```yaml
> nodeSelector:
>   kubernetes.io/hostname: <name>
> ```
> 
> Hope that clarify.

Hello @fquinino!

Thank you for the suggestion. I'll test this out.

**Comment by P4sca1** [comment]
This should be fixed with #1529

**Comment by frikky** [comment]
> This should be fixed with #1529

I trust you and @0x0elliot for this then. Thanks! :)
